beginner.md

```diff
@@ -0,0 +1,588 @@
+
+### create the scaffold
+We'll create a new `delite` custom element using Yeoman.
+
+From the command line create a new directory somewhere (named `blogging-package`) and change directory to it using the commands:
+
+    mkdir -p blogging-package
+    cd blogging-package
+
+Run Yeoman to create our scaffold.
+
+    yo delite-element
+
+You'll be prompted to enter the widget package name and the name of the custom widget element. Set the following choices shown in brackets below.
+
+    ? What is the name of your delite widget element package? (blogging-package)
+    ? What do you want to call your delite widget element (must contain a dash)? (blog-post)
+    ? Would you like your delite element to be built on a template? (Y)
+    ? Would you like your delite element to providing theming capabilities? (N)
+    ? Will your delite element require string internationalization? (N)
+    ? Will your delite element require pointer management? (N)
+    ? Do you want to use build version of delite package (instead of source version)? (N)
+
+### What's been generated
+Yeoman created the following (as shown in the console output):
+
+- `./BlogPost.js` - __this is our widget module__
+- `./BlogPost/css/BlogPost.css` - __this is our widget css__
+- `./BlogPost/BlogPost.html` - __this is our widget template__
+- `./samples/BlogPost.html` - __this is a sample how to use our new widget__
+
+You can view the sample generated HTML `./samples/BlogPost.html` in a browser to see what's been created.
+
+
+## Creating a custom element
+Viewing the `./samples/BlogPost.html` example HTML we can see we've (partly) created the custom element declaratively in markup via:
+```html
+<blog-post id="element" value="The Title"></blog-post>
+```
+For those who used the Dojo Toolkit Dijit framework previously, an important conceptual difference in `delite` is that the widget is the DOM node.
+Dijit widgets instead had a property which referenced the DOM node.
+
+For example if you open your browser developer tools and in the console enter `myvar = document.getElementById('element')` and then explore
+the properties on that variable `myvar`, you'll see it's just a regular HTML element;
+if you're more inquisitive you might be able to see there are extra properties/methods on this element which is what the `delite` framework is providing.
+
+###Registering
+
+The `<blog-post>` element doesn't constitute a custom element on its own; it first needs to go through a registration process which is achieved using
+the `delite/register` module. This is analogous to the HTML specification for registering custom elements
+i.e. `document.registerElement('blog-post');`
+
+If we look at the custom element module `./BlogPost.js` we see that we register the custom element tag via the `return register(....)` method:
+
+```js
+
+define([
+	"delite/register",
+	"delite/Widget",
+	"delite/handlebars!./BlogPost/BlogPost.html",
+    "requirejs-dplugins/css!./BlogPost/css/BlogPost.css"
```

Comment:
Looks like indentation is not the same for the last required module than others.